docs/server/env-vars.rst

环境变量
================

CSHEET_ENV

  运行环境, "development" 或者 "production"。

CSHEET_ADDRESS

  服务运行地址。默认：localhost:80

CSHEET_DATA_PATH

  :doc:`file-store` 文件夹路径，默认为工作目录下 storage 文件夹。

CSHEET_FILE_LIFE

  生成的文件保留时间, 默认为 240 小时。

  使用 go 时间格式，例如：1h2m3s。

CSHEET_TEMP_LIFE

  残留的临时文件保留时间, 默认为 1 小时。

  使用 go 时间格式，例如：1h2m3s。

CSHEET_WATCH_RATE

  限制每秒钟最多检查多少文件的更新，默认值：50。

CSHEET_WATCH_WORKERS

  文件监控并发数量，默认为 8。

CSHEET_COLLECT_IGNORE

  收集文件路径正则表达式忽略规则，多个值用换行符分割。

CSHEET_CGTEAMWORK_MAX_TASK_PER_COLLECT

  每次从 CGTeamwork 收集时的最大任务数量。默认：1000。

CSHEET_CGTEAMWORK_PIPELINE_OVERWRITE

  覆盖流程收集时使用的流程名称，JSON 编码的字典。例：``{"合成":["合成","输出"],"灯光":["灯光","渲染"]}``

CSHEET_FOLDER_INCLUDE

  从文件夹收集允许的文件夹，使用 ``*`` 代表允许任何文件夹。

  多个路径用逗号分隔。

CSHEET_FOLDER_EXCLUDE

  扫描文件夹时的前缀排除模式，用逗号分隔。

CSHEET_CLIENT_SENTRY_DSN

  客户端 Sentry 地址配置。

  后缀添加 _{客户端类型大写} 可对特定客户端指定覆盖值。

CSHEET_CLIENT_SENTRY_TRACES_SAMPLE_RATE

  客户端 Sentry 性能跟踪取样率配置，范围 0 - 1.0 。

  后缀添加 _{客户端类型大写} 可对特定客户端指定覆盖值。

CSHEET_SERVER_SENTRY_DSN

  服务端 Sentry DSN。

CSHEET_RELEASE

  发布版本，在 Sentry 事件中记录。

CSHEET_ISSUE_TRACKER_URL

  问题反馈网址，显示在客户端界面上。

CSHEET_TRANSCODE_MAX_WEIGHT

  最大的转码工作权重，同时进行的工作权重和不会超过此值，默认为 1024。

CSHEET_TRANSCODE_IMAGE_THUMB_WEIGHT

  缩略尺寸图片转码工作权重，默认为 1024 / (CPU 核数 / 2 + 1)。

CSHEET_TRANSCODE_IMAGE_REGULAR_WEIGHT

  标准尺寸图片转码工作权重，默认为 1024 / (CPU 核数 / 2 + 1)。

CSHEET_TRANSCODE_VIDEO_REGULAR_WEIGHT

  标准尺寸视频转码工作权重，默认为 1024 / 2 + 1。

CSHEET_CORS_HOSTS

  逗号分隔的主机 CORS 域名列表，例：``example.com,example.org``

  使用非标准端口时要添加端口号。

CSHEET_ADMIN_TOKEN

  调用 API 管理相关字段时使用的令牌。

  必须设置且调用时提供相同令牌才能执行相关字段。

CSHEET_DISABLE_WEBP

  禁用 WebP 图像格式，用 PNG 或者 JPG 代替。

CSHEET_API_TRACING_ENABLED

  为 API 启用 `apollo-tracing <https://github.com/apollographql/apollo-tracing>`_

CSHEET_ARCHIVE_COLLECTION_LIMIT

  限制归档打包的收藏数量，默认为 1000。

USE_X_FORWARDED_FOR

  是否使用 X-Forwarded-For 表头来记录 IP。

CGTEAMWORK_URL

  CGTeamwork 服务器地址。例：http://192.168.50.11

CGTEAMWORK_USERNAME

  CGTeamwork 用户名

CGTEAMWORK_PASSWORD

  CGTeamwork 密码


_sources/server/env-vars.rst.txt

环境变量
================

CSHEET_ENV

  运行环境, "development" 或者 "production"。

CSHEET_ADDRESS

  服务运行地址。默认：localhost:80

CSHEET_STORAGE

  :doc:`file-store` 文件夹路径。

CSHEET_FILE_LIFE

  生成的文件保留时间, 默认为 240 小时。

  使用 go 时间格式，例如：1h2m3s。

CSHEET_TEMP_LIFE

  残留的临时文件保留时间, 默认为 1 小时。

  使用 go 时间格式，例如：1h2m3s。

CSHEET_WATCH_RATE

  限制每秒钟最多检查多少文件的更新，默认值：50。

CSHEET_WATCH_WORKERS

  文件监控并发数量，默认为 8。

CSHEET_COLLECT_IGNORE

  收集文件路径正则表达式忽略规则，多个值用换行符分割。

CSHEET_CGTEAMWORK_MAX_TASK_PER_COLLECT

  每次从 CGTeamwork 收集时的最大任务数量。默认：1000。

CSHEET_CGTEAMWORK_PIPELINE_OVERWRITE

  覆盖流程收集时使用的流程名称，JSON 编码的字典。例：``{"合成":["合成","输出"],"灯光":["灯光","渲染"]}``

CSHEET_FOLDER_INCLUDE

  从文件夹收集允许的文件夹，使用 ``*`` 代表允许任何文件夹。

  多个路径用逗号分隔。

CSHEET_FOLDER_EXCLUDE

  扫描文件夹时的前缀排除模式，用逗号分隔。

CSHEET_WEB_SENTRY_DSN

  网页端 Sentry DSN。

CSHEET_SERVER_SENTRY_DSN

  服务端 Sentry DSN。

CSHEET_RELEASE

  发布版本，在 Sentry 事件中记录。

CSHEET_ISSUE_TRACKER_URL

  问题反馈网址，显示在客户端界面上。

CSHEET_TRANSCODE_MAX_WEIGHT

  最大的转码工作权重，同时进行的工作权重和不会超过此值，默认为 1024。

CSHEET_TRANSCODE_IMAGE_THUMB_WEIGHT

  缩略尺寸图片转码工作权重，默认为 1024 / (CPU 核数 / 2 + 1)。

CSHEET_TRANSCODE_IMAGE_REGULAR_WEIGHT

  标准尺寸图片转码工作权重，默认为 1024 / (CPU 核数 / 2 + 1)。

CSHEET_TRANSCODE_VIDEO_REGULAR_WEIGHT

  标准尺寸视频转码工作权重，默认为 1024 / 2 + 1。

CSHEET_CORS_HOSTS

  逗号分隔的主机 CORS 域名列表，例：``example.com,example.org``

  使用非标准端口时要添加端口号。

CSHEET_ADMIN_TOKEN

  调用 API 管理相关字段时使用的令牌。

  必须设置且调用时提供相同令牌才能执行相关字段。

CSHEET_DISABLE_WEBP

  禁用 WebP 图像格式，用 PNG 或者 JPG 代替。

CSHEET_API_TRACING_ENABLED

  为 API 启用 `apollo-tracing <https://github.com/apollographql/apollo-tracing>`_

CSHEET_ARCHIVE_COLLECTION_LIMIT

  限制归档打包的收藏数量，默认为 1000。

USE_X_FORWARDED_FOR

  是否使用 X-Forwarded-For 表头来记录 IP。

CGTEAMWORK_URL

  CGTeamwork 服务器地址。例：http://192.168.50.11

CGTEAMWORK_USERNAME

  CGTeamwork 用户名

CGTEAMWORK_PASSWORD

  CGTeamwork 密码


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

## [6.4.0](https://github.com/WuLiFang/csheet/compare/v6.3.0...v6.4.0) (2021-04-06)


### Features

* support viewer fullscreen ([c38a1eb](https://github.com/WuLiFang/csheet/commit/c38a1eb8405d02186037fcc8c14c4a46e2ddd9fc))
* **api:** add directive `ignoreError` ([dda2938](https://github.com/WuLiFang/csheet/commit/dda2938d3a0f839d7154e93edf643b08dbdbbb11))
* **archive:** include cgteamwork status color ([6f48421](https://github.com/WuLiFang/csheet/commit/6f484216ebe9f268e85c1a9c69c69df86c81d400))


### Bug Fixes

* presentation controls should not initialize before parent ([18a4403](https://github.com/WuLiFang/csheet/commit/18a440390b45497bb9a8a0765334afaecd75cf40))
* **style:** wrong annotation width for long media ([cbc6112](https://github.com/WuLiFang/csheet/commit/cbc611273a4579773593c8973d52ebc0ad270cfc))
* data path env var not work ([260b885](https://github.com/WuLiFang/csheet/commit/260b885e4ebe7835f83521ccc4e0e84b5d65a1cf))
* presentation should be draggable ([fa040f5](https://github.com/WuLiFang/csheet/commit/fa040f5ee29d6a32f223d8cfa4f8198bb947b5c1))
* wrong object contain rate when inner size is smaller ([e4e4311](https://github.com/WuLiFang/csheet/commit/e4e4311a19364b17cef951ffb6ff3cb369ffc043))
* **style:** wrong button icon size in small screen ([98cd54d](https://github.com/WuLiFang/csheet/commit/98cd54d09cbf7bb872a795e99e7b97a6b5756826))
* should not archive same file twice ([afb314e](https://github.com/WuLiFang/csheet/commit/afb314e4f1f3fb4964df4de2b50d9ae330da945a))
* should not show transcoding message when data not loaded ([b78b1cd](https://github.com/WuLiFang/csheet/commit/b78b1cd76c52a9b93a43bae4c3fab3cbd73af8e2))
* should not show transcoding when transcode failed ([c37712f](https://github.com/WuLiFang/csheet/commit/c37712f2ea469305b2f552b9e8ec55b1cc6711f5))
* should update document title on state load ([625b5ad](https://github.com/WuLiFang/csheet/commit/625b5ada57f313a7113e866df66f5cafefc7cc0b))
* should update loading count for initial loading ([152ca22](https://github.com/WuLiFang/csheet/commit/152ca2263ea21bb43417df7530be9ddaa971efe4))
* type error in PresentationControls.vue ([7f4a1a7](https://github.com/WuLiFang/csheet/commit/7f4a1a79fb2274bd208daa8f9bf5faf2c5f05dba))
* unknown task status should has higher priority than Approve/Wait ([cd01d39](https://github.com/WuLiFang/csheet/commit/cd01d39f9a4ef21cb6a91a56475a17a95dc1333c))
* wrong archive title ([246f068](https://github.com/WuLiFang/csheet/commit/246f06894c581af80316898cac71639e20ac805b))
* wrong user ip in sentry event ([cbbf683](https://github.com/WuLiFang/csheet/commit/cbbf6836288d849e4f512a7a36b124dc492ed9c2))

## [6.3.0](https://github.com/WuLiFang/csheet/compare/v6.2.0...v6.3.0) (2021-03-30)

### Features

- support page archive download ([8d3911a](https://github.com/WuLiFang/csheet/commit/8d3911a956c6ec9785d7b8d3e84f9c75ae1f4b33))
- **style:** show scroll bar for long media ([002043e](https://github.com/WuLiFang/csheet/commit/002043e662763b4e1a0bf9a6181c3fb21d17d472))
- support collect ignore ([43b00ec](https://github.com/WuLiFang/csheet/commit/43b00ec670b280b2f472bf8b132efff24ca58f28))

### Bug Fixes

- missing frame count metadata in viewer ([bc88387](https://github.com/WuLiFang/csheet/commit/bc883874ecc2be3a00148452b2adc19a8e3b8cb9))
- **deps:** go mod tidy ([fbd94c0](https://github.com/WuLiFang/csheet/commit/fbd94c03dabb98d653e4cb46101f374853de5498))
- **style:** annotation not work for long media ([9c53f84](https://github.com/WuLiFang/csheet/commit/9c53f843cf170cf24087d3c0fe8b1cbe74fb2ca4))
- **style:** should not show presentation scroll bar during loading ([73c0885](https://github.com/WuLiFang/csheet/commit/73c0885ab0c7b465c9aa923ad1992f3e3e002e2f))
- handle broken files that can not probe ([b69d394](https://github.com/WuLiFang/csheet/commit/b69d3946c27fe8afd36529bb443a5486c63c14e9))
- **deps:** update dependency vue-i18n to v8.24.0 ([8a20386](https://github.com/WuLiFang/csheet/commit/8a20386daa9d254979d79b49115735f428e56875))

## [6.2.0](https://github.com/WuLiFang/csheet/compare/v6.1.0...v6.2.0) (2021-03-08)

### Features

- hide selected option in collection tag input ([ba2f783](https://github.com/WuLiFang/csheet/commit/ba2f783bac22cf76d269ecdbb033aa6244609a99))
- not allow create tag in collection filter ([049a558](https://github.com/WuLiFang/csheet/commit/049a558f178b920f588a063f845ded08d69c563e))

### Bug Fixes

- long click in dropdown not work ([7a4b913](https://github.com/WuLiFang/csheet/commit/7a4b913c086021f97f981e5996a6e99cd988432a))

## [6.1.0](https://github.com/WuLiFang/csheet/compare/v6.0.0...v6.1.0) (2021-03-02)

### Features

- handle recreated cgteamwork task ([c6e7a4e](https://github.com/WuLiFang/csheet/commit/c6e7a4e7d88c3436ca630f13bb07c43ad04dcfe6))
- **style:** add transition to collection tag treemap ([5dc77fe](https://github.com/WuLiFang/csheet/commit/5dc77febe69552912ee8db0932ae406d1a4d2b22))
- **style:** add tween for collection tag count in treemap ([77c107d](https://github.com/WuLiFang/csheet/commit/77c107d9f78386d17152abca11cfb0319ad55ce2))
- add cgteamwork checkbox auto tag ([ec7d8e2](https://github.com/WuLiFang/csheet/commit/ec7d8e2033f27aaece7093a86597c3b00c9cf85c))
- add loading indicator to cgtw inputs ([c74da2d](https://github.com/WuLiFang/csheet/commit/c74da2d663fdcd3ca984bf829d8c4df8b140451d))
- add treemap to collection tag stats ([c214582](https://github.com/WuLiFang/csheet/commit/c214582f138fd127a505fdd4113a4c8b27aef625))
- allow change args in collection stats drawer ([0cf9e04](https://github.com/WuLiFang/csheet/commit/0cf9e044275c0ed878bb84cda176b10b173b3bda))
- change cgteamwork auto tag format ([4f7a41a](https://github.com/WuLiFang/csheet/commit/4f7a41ac9cbfef68d2ecd9abcaac4372e27d9476))

internal/config/config.go
package config

import (
	"os"
	"strings"
	"time"

	"github.com/NateScarlet/zap-sentry/pkg/logging"
	"github.com/WuLiFang/csheet/v6/pkg/getenv"
	"github.com/WuLiFang/csheet/v6/pkg/job/transcode"
)

func newStringSet(v []string) (ret map[string]struct{}) {
	ret = make(map[string]struct{})
	for _, i := range v {
		ret[i] = struct{}{}
	}
	return
}

// Config entries
// TODO: change default data path to `data` and drop old config support, this will be a breaking change
var (
	Env                         = getenv.String("CSHEET_ENV", "development")
	Address                     = getenv.String("CSHEET_ADDRESS", "0.0.0.0:80")
	DataPath                    = getenv.StringCoalesce([]string{"CSHEET_DATA_PATH", "CSHEET_STORAGE"}, "storage")
	WatchRate                   = getenv.Int("CSHEET_WATCH_RATE", 50)
	FileLife                    = getenv.Duration("CSHEET_FILE_LIFE", 240*time.Hour)
	TempLife                    = getenv.Duration("CSHEET_TEMP_LIFE", time.Hour)
	CollectIgnore               = getenv.String("CSHEET_COLLECT_IGNORE", "")
	FolderInclude               = getenv.String("CSHEET_FOLDER_INCLUDE", "*")
	FolderExclude               = getenv.String("CSHEET_FOLDER_EXCLUDE", "")
	CGTeamworkMaxTaskPerCollect = getenv.Int("CSHEET_CGTEAMWORK_MAX_TASK_PER_COLLECT", 1000)
	CGTeamworkPipelineOverwrite = getenv.String("CSHEET_CGTEAMWORK_PIPELINE_OVERWRITE", "{}")
	SentryDSN                   = getenv.String("CSHEET_SERVER_SENTRY_DSN", "")
	Release                     = getenv.String("CSHEET_RELEASE", "")
	IssueTrackerURL             = getenv.String("CSHEET_ISSUE_TRACKER_URL", "https://github.com/WuLiFang/csheet/issues")
	WatchWorkers                = getenv.Int("CSHEET_WATCH_WORKERS", 8)
	TranscodeMaxWeight          = getenv.Int64("CSHEET_TRANSCODE_MAX_WEIGHT", transcode.MaxWeight)
	TranscodeImageThumbWeight   = getenv.Int64("CSHEET_TRANSCODE_IMAGE_THUMB_WEIGHT", transcode.ImageThumbWeight)
	TranscodeImageRegularWeight = getenv.Int64("CSHEET_TRANSCODE_IMAGE_REGULAR_WEIGHT", transcode.ImageRegularWeight)
	TranscodeVideoRegularWeight = getenv.Int64("CSHEET_TRANSCODE_VIDEO_REGULAR_WEIGHT", transcode.VideoRegularWeight)
	CORSHosts                   = strings.Split(getenv.String("CSHEET_CORS_HOSTS", ""), ",")
	AdminToken                  = getenv.String("CSHEET_ADMIN_TOKEN", "")
	DisableWebP                 = getenv.Bool("CSHEET_DISABLE_WEBP", false)
	APITracingEnabled           = getenv.Bool("CSHEET_API_TRACING_ENABLED", false)
	ArchiveCollectionLimit      = getenv.Int("CSHEET_ARCHIVE_COLLECTION_LIMIT", 1000)
	UseXForwardedFor            = getenv.Bool("USE_X_FORWARDED_FOR", false)
)

func init() {
	logger := logging.Logger("config")
	if v := os.Getenv("CSHEET_WEB_SENTRY_DSN"); v != "" {
		if os.Getenv("CSHEET_CLIENT_SENTRY_DSN") == "" {
			logger.Warn("CSHEET_WEB_SENTRY_DSN is depreacted, use CSHEET_CLIENT_SENTRY_DSN instead")
			os.Setenv("CSHEET_CLIENT_SENTRY_DSN", v)
		} else {
			logger.Warn("ignore CSHEET_WEB_SENTRY_DSN when CSHEET_CLIENT_SENTRY_DSN set")
		}
	}
	if v := os.Getenv("CSHEET_STORAGE"); v != "" {
		logger.Warn("CSHEET_STORAGE is depreacted, use CSHEET_DATA_PATH instead")
	}
}
